### DISCUSSION TITLE
snake_case

### DISCUSSION THREAD

**Discussion by AndriiDidkivsky** [vlang/v#9166] [General]
Hello i'm getting json from third-party service and get next error during compilation. 
`field name containerType cannot contain uppercase letters, use snake_case instead`
so i have to map literally every  json with camelCase into v's structure with snake_case. 
What's the reason to this restriction ?

**Comment by medvednikov** [comment]
yeah it's annoying

we'll most likely make the json decoder smarter and assign fieldName to field_name

**Reply by AndriiDidkivsky** [reply]
ok, but what about encoding. 
Sometimes we need to convert  snake_case to camelCase during serizlization sometimes no. I have no clue how to do it correctly now. Of course I can write custom serializer or implement `json2` interface but i have to do it for every Entity. Looks weird :\

**Reply by theoparis** [reply]
> ok, but what about encoding. Sometimes we need to convert snake_case to camelCase during serizlization sometimes no. I have no clue how to do it correctly now. Of course I can write custom serializer or implement `json2` interface but i have to do it for every Entity. Looks weird :\

I have the same issue, I need to interface with the podman rest api, however I need the json encoding and decoding to be case insensitive due to the api using PascalCase.

**Reply by medvednikov** [reply]
you can do `field string [json:'MyField']`

**Reply by theoparis** [reply]
That may work but does not solve the issue of having to put that on each field in each struct.

**Reply by JalonSolov** [reply]
Only when the field name doesn't already match the json key (in case as well as spelling, etc.).  Go does the same thing, but it's been around long enough there are numerous ways to generate the struct from the json, such as https://mholt.github.io/json-to-go/ - it would be nice if there were a `json2v`.

**Comment by medvednikov** [comment]
snake_case is enforced for consistency across all code bases
